### PR TITLE
Add GitLab-CI yaml

### PR DESCRIPTION
**Pull request by SuperQ**
Add a GitLab CI configuration for building artifacts on GitLab.

### CODE REVIEW

**Comment by jaqx0r** [comment]
Can you elaborate please, what does adding gitlab do and how do we use it?
Assume I am a total noob.

**Comment by SuperQ** [comment]
This allows any gitlab users to have an automatic binary build pipeline.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  stage: test
+  allow_failure: true
+  script:
+    - make test
+
+build:
+  stage: build
+  script:
+    - make install
+    - go build -o ${CI_PROJECT_DIR}/build/mtail
```
Comment (jaqx0r) [inline .gitlab-ci.yml]: I don't understand how this used, why not copy from $GOPATH/bin/mtail ?  Is this really masking a bug in the install?

Comment (SuperQ) [inline .gitlab-ci.yml]: Hrmm, good question, I didn't actually write this.  I just copied what we had [upstream](https://gitlab.com/gl-infra/mtail/blob/master/.gitlab-ci.yml).

I will look into why this is.

**Comment by SuperQ** [comment]
Tested the config upstream with a `cp` instead of a build.  Works perfectly.

**Pull request merged**